pages/dao/[symbol]/proposal/components/CancelProposal.tsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, { FunctionComponent } from 'react'
+import { RpcContext } from 'models/core/api'
+import useWalletStore from 'stores/useWalletStore'
+import useRealm from 'hooks/useRealm'
+import Button from '@components/Button'
+import { notify } from 'utils/notifications'
+import Modal from '@components/Modal'
+import { Proposal } from 'models/accounts'
+import { ParsedAccount } from 'models/core/accounts'
+import { cancelProposal } from 'actions/cancelProposal'
+import useProposal from '@hooks/useProposal'
+
+interface CancelProposalModalProps {
+  onClose: () => void
+  isOpen: boolean
+}
+
+const CancelProposalModal: FunctionComponent<CancelProposalModalProps> = ({
+  onClose,
+  isOpen,
+}) => {
+  const wallet = useWalletStore((s) => s.current)
+  const connection = useWalletStore((s) => s.connection)
+  const { realmInfo } = useRealm()
+
+  const { proposal } = useProposal()
+
+  const rpcContext = new RpcContext(
+    proposal!.account.owner,
+    realmInfo?.programVersion,
+    wallet,
+    connection.current,
+    connection.endpoint
+  )
+
+  const handleCancelProposal = async (
+    proposal: ParsedAccount<Proposal> | undefined
+  ) => {
+    try {
+      await cancelProposal(rpcContext, proposal)
+
+      onClose()
+    } catch (error) {
+      notify({
+        type: 'error',
+        message: `Error: Could not cancel proposal.`,
+      })
+
+      onClose()
+
+      console.log('error cancelling proposal', error)
+    }
+  }
+
+  return (
+    <Modal onClose={onClose} isOpen={isOpen}>
+      <h2>Cancel proposal</h2>
+
+      <p className="text-left mt-5 mb-8">
+        Do you want to cancel this proposal?
+      </p>
+
+      <div className="flex items-center justify-center">
+        <Button className="w-44 mr-4" onClick={onClose}>
```

Comment:
No, should be secondary button (not primary) in all modals

Reply:
Only for the guarantee, are you sure that this is a safe choice? Bootstrap standardizes `Ok` to the right and `Denial/Cancel` to the left and most of the UI frameworks and sites use this model. This could cause misclicks

Reply:
By `secondary` I meant the  style of the button  and not its position. Currently the No buttons  have `primary` action style